streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/TableSourceNode.java

```diff
@@ -103,8 +109,13 @@ public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
             // only add state store if the source KTable should be materialized
             final KTableSource<K, V> ktableSource = (KTableSource<K, V>) processorParameters.processorSupplier();
             if (ktableSource.queryableName() != null) {
-                topologyBuilder.addStateStore(storeBuilder, nodeName());
-                topologyBuilder.markSourceStoreAndTopic(storeBuilder, topicName);
+                if (shouldReuseSourceTopicForChangelog) {
+                    storeBuilder.withLoggingDisabled();
+                    topologyBuilder.addStateStore(storeBuilder, nodeName());
+                    topologyBuilder.connectSourceStoreAndTopic(storeBuilder.name(), topicName);
+                } else {
+                    topologyBuilder.addStateStore(storeBuilder, nodeName());
```

Comment:
nit: we can move this line up outside the if condition and remove else?

Reply:
Ack